### PR TITLE
bugfix for gps tracking not working in containers

### LINKED ISSUE #1173: GPS in mining sub not working
[BUG]

**Describe the bug**
While in a mining sub, the space gps item does not work. Not sure if this is true for all subs

**To Reproduce**
1. Get Space GPS
2. Go to trench
3. Get in sub
4. Try to input GPS coordinates

**Expected behavior**
GPS coordinated entered for tracking

### PR DESCRIPTION
**Pull request by Adharainspace**
[FEAT]
## About the PR 
some weird code was making it so that if you tried to track a location with your gps while in a pod, sub, or locker, it just sent an error message when it shouldntve. this fixes that!

## Why's this needed? 
Fixes #1173 bugs bad

## Changelog

```
(u)adharainspace:
(+)GPS tracking no longer broken while inside pods/subs.
```

### CODE REVIEW

**Comment by ZeWaka** [comment]
I don't see why the locate() part and T.z part is necessary at all.

**Comment by Adharainspace** [comment]
> I don't see why the locate() part and T.z part is necessary at all.
the way the gps pointer works, it needs a turf to point at, so i use locate with x, y and z coordinates (since the gps takes in x and y coordinates, and we'll be pointing at something on the same z level as us)

**Comment by ZeWaka** [comment]
> 
> 
> > I don't see why the locate() part and T.z part is necessary at all.
> 
> the way the gps pointer works, it needs a turf to point at, so i use locate with x, y and z coordinates (since the gps takes in x and y coordinates, and we'll be pointing at something on the same z level as us)

no but like, you already do `var/turf/T = get_turf(src)` - you already have the turf reference

File: code/obj/item/device/gps.dm [hunk 1]
```diff
@@ -219,11 +219,10 @@ var/global/list/all_GPSs = list()
 			if (!x || !y)
 				boutput(usr, "<span class='alert'>Bad Topc call, if you see this something has gone wrong. And it's probably YOUR FAULT!</span>")
 				return
-			var/z = src.z
-			if (src.loc)
-				z = src.loc.z
+			var/turf/T = get_turf(src) //bugfix for this not working when src was in containers
+			var/z = T.z
```
Comment (ZeWaka) [inline code/obj/item/device/gps.dm:224]:
```suggestion
			// This is to get a turf with the specified coordinates on the same Z as the device
			var/turf/T = get_turf(src) //bugfix for this not working when src was in containers
			var/z = T.z

```

File: code/obj/item/device/gps.dm [hunk 2]
```diff
@@ -219,11 +219,10 @@ var/global/list/all_GPSs = list()
...
```
Comment (ZeWaka) [inline code/obj/item/device/gps.dm:223]:
```suggestion
			// This is to get a turf with the specified coordinates on the same Z as the device
			var/turf/T = get_turf(src) //bugfix for this not working when src was in containers
			var/z = T.z

```

Comment (Adharainspace) [reply code/obj/item/device/gps.dm:223]: looks good to me, comments are good

**Comment by Adharainspace** [comment]
will add suggestion when i wake up, idk how it works on mobile

**Comment by Adharainspace** [comment]
done! thanks zewaka :]

**Pull request merged**